specs/invalid-tx-proof.md

```diff
@@ -0,0 +1,67 @@
+# Invalid transaction proof
+
+## Type of invalid transaction
+According to [Taiko whitepaper](https://github.com/taikochain/whitepaper), we still generate proof for invalid transactions. A TX is invalid means it dis-obeys the follow rules come from ethereum yellow paper. 
+
+```
+6. Transaction Execution
+The execution of a transaction is the most complex part of the Ethereum protocol: it defines the state transition function Υ. It is assumed that any transactions executed first pass the initial tests of intrinsic validity. These include:
+(1) The transaction is well-formed RLP, with no additional trailing bytes;
+(2) the transaction signature is valid;
+(3) the transaction nonce is valid (equivalent to the sender account’s current nonce);
+(4) the sender account has no contract code deployed (see EIP-3607 by Feist et al. [2021]);
+(5) the gas limit is no smaller than the intrinsic gas, g0, used by the transaction; and
+(6) the sender account balance contains at least the cost, v0, required in up-front payment.
+```
+
+However we don't check all of them in evm_circuit. Some checks can be done in the contract. So we separate TX to these 3 types:
+1. `Good TX`: A TX can be executed successfully, which means it comply with that 6 rules defined by the eth yellow paper.
+2. `Neutral bad TX`: A bad TX, but the failure is becuase sth out of a miner's control, e.g., a nonce conflicting TX.
+3. `Malicious bad TX`: A bad TX, but it is submitted because of a malicious proposer. For example: a TX with wrong signature, normal miner filter it out. The txList contains TX in this type is also treated as invalid txList.
```

Comment:
Change `as invalid txList.` to `as an invalid txList.`?

Reply:
Fixed